modules/actions/snapshots/sync-snapshots.ts

```diff
@@ -0,0 +1,119 @@
+import { Chain, Prisma, PrismaLastBlockSyncedCategory } from '@prisma/client';
+import { prisma } from '../../../prisma/prisma-client';
+import _ from 'lodash';
+import { PoolSnapshot_Filter } from '../../sources/subgraphs/balancer-v3-vault/generated/types';
+import { fillMissingTimestamps } from './lib/fill-missing-timestamps';
+import { computeDailyValues } from './lib/compute-daily-values';
+import { applyUSDValues } from './lib/apply-usd-values';
+
+type SnapshotsSubgraphClient = {
+    getMetadata: () => Promise<{ block: { number: number } }>;
+    getAllSnapshots: (args: PoolSnapshot_Filter) => Promise<Prisma.PrismaPoolSnapshotUncheckedCreateInput[]>;
+};
+
+export async function syncSnapshots(
+    subgraphClient: SnapshotsSubgraphClient,
+    category: PrismaLastBlockSyncedCategory,
+    chain: Chain,
+    options: {
+        poolIds?: string[]; // Optional: Specific pool IDs to sync
+        syncLatest?: boolean; // Optional: Whether to sync based on the latest block
```

Comment:
this is a bit bogus, would maybe switch the meaning and call it `syncFromScratch` or `syncFromInception`

Reply:
`fullSync`?

Reply:
although reverse boolean makes it harder to read later on when if branching happens

Reply:
startFromLastSyncedBlock